Command/BackupCommand.php

```diff
@@ -55,116 +55,130 @@ protected function initialize(InputInterface $input, OutputInterface $output)
+                }
+                else{
+                    $this->getContainer()->get('dizda.cloudbackup.client.dropbox')->upload($wholeFile);
+                }
             }
-        }
 
-        if (isset($this->storages['cloudapp'])) {
-            if(in_array('cloudapp', $this->splitStorages)){
-                $this->getContainer()->get('dizda.cloudbackup.client.cloudapp')->upload($splitFiles);
+            if (isset($this->storages['google_drive'])) {
+                if(in_array('google_drive', $this->splitStorages)){
+                    $this->getContainer()->get('dizda.cloudbackup.client.google_drive')->upload($splitFiles);
+                }
+                else{
+                    $this->getContainer()->get('dizda.cloudbackup.client.google_drive')->upload($wholeFile);
+                }
             }
-            else{
-                $this->getContainer()->get('dizda.cloudbackup.client.cloudapp')->upload($wholeFile);
-            }
-        }
 
-        if (isset($this->storages['gaufrette'])) {
-            $filesystemName = $this->getContainer()->getParameter('dizda_cloud_backup.cloud_storages')['gaufrette']['service_name'];
+            if (isset($this->storages['cloudapp'])) {
+                if(in_array('cloudapp', $this->splitStorages)){
+                    $this->getContainer()->get('dizda.cloudbackup.client.cloudapp')->upload($splitFiles);
+                }
+                else{
+                    $this->getContainer()->get('dizda.cloudbackup.client.cloudapp')->upload($wholeFile);
+                }
+            }
 
-            $gaufrette = $this->getContainer()->get('dizda.cloudbackup.client.gaufrette');
-            $gaufrette->setFilesystem($this->getContainer()->get($filesystemName));
-            if(in_array('gaufrette', $this->splitStorages)){
-                $gaufrette->upload($splitFiles);
+            if (isset($this->storages['gaufrette'])) {
+                $filesystemName = $this->getContainer()->getParameter('dizda_cloud_backup.cloud_storages')['gaufrette']['service_name'];
+
+                $gaufrette = $this->getContainer()->get('dizda.cloudbackup.client.gaufrette');
+                $gaufrette->setFilesystem($this->getContainer()->get($filesystemName));
+                if(in_array('gaufrette', $this->splitStorages)){
+                    $gaufrette->upload($splitFiles);
+                }
+                else{
+                    $gaufrette->upload($wholeFile);
+                }
             }
-            else{
-                $gaufrette->upload($wholeFile);
+
+            $processor->cleanUp();
+            $this->output->writeln('- <comment>Temporary files have been cleared</comment>.');
+        }catch(\Exception $e){  
+            foreach($this->getContainer()->getParameter('dizda_cloud_backup.error_notification')['to'] as $to){
+                $message = $this->getContainer()->get('mailer')->createMessage()
```

Comment:
Before sending the notification, you should test if the parameter 'error_notification' is set.
Otherwise if the user did not configurated the error_notification, it could crash when an Exception happen.